lib/src/inline_syntaxes/decode_html_syntax.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../assets/html_entities.dart';
+import '../ast.dart';
+import '../charcode.dart';
+import '../inline_parser.dart';
+import '../patterns.dart';
+import '../util.dart';
+import 'inline_syntax.dart';
+
+/// Decodes numeric character references, for example decode `&#35;` to `#`.
+// https://spec.commonmark.org/0.30/#entity-and-numeric-character-references
+class DecodeHtmlSyntax extends InlineSyntax {
+  DecodeHtmlSyntax()
+      : super(htmlCharactersPattern.pattern,
+            caseSensitive: false, startCharacter: $ampersand);
+
+  @override
+  bool tryMatch(InlineParser parser, [int? startMatchPos]) {
+    if (parser.pos > 0 && parser.charAt(parser.pos - 1) == $backquote) {
+      return false;
+    }
+    final match = pattern.matchAsPrefix(parser.source, parser.pos);
+    if (match == null ||
+        match[1] != null ||
+        (match[2] == null && match[3] == null)) {
```

Comment:
This is rather inscrutible, haha. Would you mind adding a comment, or creating local variables with meaningful names (e.g. for `match[1]` which is recalled on line 32).

Reply:
Yep, it is confusing. 

I have simplified the `if` condition. 

I guess I was trying to temporarily disable HTML entity decoding but keep most of the logic, so I can  easily enable it again later when dealing with HTML escaping issues.

```dart
match[1] != null;  // it  is an HTML entity reference
match[2] != null;  // it is a decimal numeric character references 
match[3] != null;  // it is a hexadecimal numeric character references
```